src/transformers/models/t5/modeling_t5.py

```diff
@@ -915,6 +910,9 @@ def forward(
                     encoder_extended_attention_mask = encoder_extended_attention_mask.to(hidden_states.device)
                 if encoder_decoder_position_bias is not None:
                     encoder_decoder_position_bias = encoder_decoder_position_bias.to(hidden_states.device)
+                if all_hidden_states is not None:
+                    all_hidden_states = all_hidden_states.to(hidden_states.device)
+
```

Comment:
This on the other hand needs to be switched. Otherwise we end up with the next line where 2 vars could be on 2 different devices.

Moreover it's telling me, there is a test missing for this case.